datafusion/optimizer/src/analyzer/type_coercion.rs

```diff
@@ -86,6 +87,18 @@ fn analyze_internal(
     external_schema: &DFSchema,
     plan: LogicalPlan,
 ) -> Result<Transformed<LogicalPlan>> {
+    if let LogicalPlan::Filter(Filter {
```

Comment:
I think this works and handles the special case reasonably

Another way that might be more general purpose would be to coerce the argument to boolean with https://docs.rs/datafusion/latest/datafusion/logical_expr/type_coercion/binary/fn.comparison_coercion.html

SO something like 
```rust
let predicate = comparison_coercion(DataType::Binary, predicate.data_type())?;
```

Reply:
If you do this case, you'll have to allow the rest of the coercion proceed

Reply:
Thanks :) Done